File: frontend/src/types/models/graph.ts


export interface IGraphData {
	modularity? : number ,
	nodes : {
		id : string ,
		pos? : string ,
		betweenness? : number
		page_rank? : number
		freq? : number
		gatewayness? : number
		modularity_community? : number
	}[],
	edges : {
		source : string ,
		target : string ,
		weight? : number ,
		pmi? : number ,
	}[],
}


export type IGraph = {
	id :string ,
	conversation_id : string ,
	graph_data? : IGraphData ,
	created_at : string ,
	udpated_at : string ,
	deleted_at : string ,
}

File: backend/internal/app/api/graph/graph.go
package graph

import "time"



type Node struct {
	ID string `json:"id"`
    ModularityCommunity int `json:"modularity_community"`
	Freq int `json:"freq"`
    Pos string `json:"pos"`
	Betweenness float32 `json:"betweenness"`
	Gatewayness float32 `json:"gatewayness"`
    PageRank float32 `json:"page_rank"`
}


type Edge struct {
	Source string `json:"source"`
	Target string `json:"target"`
	Weight int    `json:"weight"`
	Pmi float32    `json:"pmi"`
}




type GraphData struct {
	Modularity float32 `json:"modularity"`
    Nodes []Node `json:"nodes"`
    Edges []Edge `json:"edges"`
}




type Graph struct {
	Id             string     `json:"id"`
	ConversationId string     `json:"conversation_id"`
	Data           GraphData     `json:"data"`
	CreatedAt      *time.Time `json:"created_at,omitempty"`
	UpdatedAt      *time.Time `json:"updated_at,omitempty"`
	DeletedAt      *time.Time `json:"deleted_at,omitempty"`
}


